### PR TITLE
contracts: Don't put unstable functions in special module

### PR DESCRIPTION
**Pull request by athei**
When we introduced the [unstable interfaces](https://github.com/paritytech/substrate/tree/master/frame/contracts#unstable-interfaces) feature we decided to put those features into a special module called `__unstable__`. This turned out to be a mistake: Whenever we eventually device to stabilize a function we generate a breaking change where a newer `pallet-contracts`version gets incompatible with an older ink!.

This is a big no go for stable functions but is allowed for unstable functions. That said, changing the module on stabilization is just a unnecessary form of breakage. Most of the time functions are stabilized as is.

This PR changes the behavior to to make version and instability orthogonal. The way we want to use this it to put functions into the module in which they are eventually stabilized (if they are).

### CODE REVIEW

File: frame/contracts/proc-macro/src/lib.rs [hunk 1]
```diff
@@ -496,31 +509,36 @@ fn expand_functions(
...
 /// appropriate attribute as follows:
 ///
 /// ## Example
 ///
 /// ```nocompile
 /// #[define_env]
 /// pub mod some_env {
-/// 	#[version(1)]
-/// 	fn some_host_fn(ctx: _, memory: _, key_ptr: u32, value_ptr: u32, value_len: u32) -> Result<ReturnCode, TrapReason> {
+/// 	#[version(0)]
```
Comment (agryaznov) [inline frame/contracts/proc-macro/src/lib.rs:528]:
```suggestion
/// 	#[version(2)]
```

File: frame/contracts/README.md [hunk 2]
```diff
@@ -142,7 +142,7 @@ this pallet contains the concept of an unstable interface. Akin to the rust nigh
 it allows us to add new interfaces but mark them as unstable so that contract languages can
 experiment with them and give feedback before we stabilize those.
 
-In order to access interfaces marked as `__unstable__` in `runtime.rs` one need to compile
+In order to access interfaces marked as `#[unstable]` in `runtime.rs` one need to compile
 this crate with the `unstable-interface` feature enabled. It should be obvious that any
 live runtime should never be compiled with this feature: In addition to be subject to
 change or removal those interfaces do not have proper weights associated with them and
```
Comment (agryaznov) [inline frame/contracts/README.md:148]: why not proper weights? we do have benches for the unstable functions and therefore have them weighted, don't we?

**Review by agryaznov — APPROVED** [review]
Just some nit picks to docs. Also, we'll certainly need to make a follow-up to ink! for this PR.

File: frame/contracts/README.md (hunk 2 above)
Comment (athei) [reply frame/contracts/README.md:148]: Right now yes. But in general we want to reserve ourselves the right to just quickly draft up a new function for testing without worrying about benchmarks.

**Comment by athei** [comment]
> Also, we'll certainly need to make a follow-up to ink! for this PR.

Yes. But only for the `take_storage` API I think. The other stuff is not yet implemented in ink! anyways.

**Review by xermicus — APPROVED** [review]
Nice, I like this approach

**Pull request merged**